src/hotspot/share/jeandle/jeandleAbstractInterpreter.hpp

```diff
@@ -38,6 +38,32 @@
 #include "memory/allocation.hpp"
 #include "utilities/bitMap.inline.hpp"
 
+/* A pair of BasicType and llvm::Value* used by JeandleVMState */
+class TypedValue {
+private:
+  BasicType _basic_type;
+  llvm::Value * _value;
+public:
+  TypedValue(BasicType type, llvm::Value* value) : _basic_type(type), _value(value) {}
```

Comment:
We can add an assertion here to make sure the `type` and the type of `value` are compatible.